dev/Tiles/metatile.h

```diff
@@ -0,0 +1,60 @@
+#include <stdbool.h>
+#include <stdlib.h>
+#include "../../lib/SMSlib.h"
+#include "tile.h"
+
+typedef struct MetaTile
+{
+    struct Tile tiles[16];
+    unsigned char numTiles;
+    unsigned char numTilesX;
+    unsigned char numTilesY;
+} MetaTile;
+
+MetaTile* MetaTile_Create
+(
+    Tile tiles[16],
+    unsigned char numTiles,
+    unsigned char numTilesX,
+    unsigned char numTilesY
+) {
+    unsigned char tileXY = numTilesX * numTilesY;
+
+    if(tileXY > 16)
+    {
+        // We have an error here
+    }
+    unsigned char numMetatiles = numTiles / tileXY;
+    
+    MetaTile* metatile = malloc(sizeof (struct MetaTile));
+    metatile->numTilesX = numTilesX;
+    metatile->numTilesY = numTilesY;
+    metatile->numTiles = numTiles;
+
+    unsigned char i;
+    for(i = 0; i < metatile->numTiles; ++i)
+    {
+        metatile->tiles[i] = tiles[i];
+    }
+
+    return metatile;
+}
+
+void MetaTile_Draw
+(
+    MetaTile* metatile,
+    unsigned char x,
+    unsigned char y
+) {
+    unsigned int tileIds[16];
+    unsigned char tilesXY = metatile->numTilesX;
+
+    unsigned char i;
+    for(i = 0; i < tilesXY; ++i)
+    {
+        tileIds[i] = metatile->tiles[0].computedId; // TODO: Not working correctly
+        tileIds[i] = 0x1011;
+    }
+
+    SMS_loadTileMapArea(x, y, tileIds, 2, 2);
```

Comment:
Hang on, are you only copying 4 tiles here? If so, why do you need 16 tile ids?

Something about this function smells a bit off, although at the risk of mixing metaphors, I can't quite put my finger on it. It does seem a little unnecessary to spin up an array of tile ids only to pass that into SMSLib to copy to the VDP, although SMSLib doesn't expose many low level functions.

One way you could do it if you wanted to optimise this function is to loop over the height of your meta tile and use a combination of `SMS_setNextTileatXY` and `SMS_setTile` to stream the tile data.